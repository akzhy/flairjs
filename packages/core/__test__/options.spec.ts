import { describe, expect, test } from 'vitest'

import { readFileSync } from 'node:fs'
import path from 'node:path'
import { dirname } from 'path'
import { fileURLToPath } from 'url'
import { transformCode } from '../index'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)

const classNameListContent = readFileSync(path.resolve(__dirname, './snippets/options-class-name-list.tsx'), 'utf-8')

describe('Options tests', () => {
  test('class name list is working', () => {
    const regex = new RegExp("^class[A-Z][A-Za-z0-9_]*");
    const result = transformCode(classNameListContent, 'options-1.tsx', {
      classNameList: ['className', 'containerClassName', regex.toString()],
      cssOutDir: path.resolve(__dirname, './.css'),
    })
    if (!result) {
      throw new Error('transformCode returned null or undefined')
    }
    expect(result.code).toMatchSnapshot()
  })
})
