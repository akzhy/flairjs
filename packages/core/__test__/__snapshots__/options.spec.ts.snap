// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Options tests > class name list is working 1`] = `
"import "@flairjs/css/cached-css/6091365a83003455.css";
import clsx from "clsx";
import { flair } from "@flairjs/react";
export const TestCaseComponent = () => {
	return <>
      <div>
        <Box containerClassName="Fo9SUq_item"></Box>
        <RegexBox classList={["Fo9SUq_regex-box"]} />
      </div>
    </>;
};
const Box = ({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) => {
	return <div className={clsx("etqoJq_box", containerClassName)}>{children}</div>;
};
const RegexBox = ({ children, classList }: {
	children?: React.ReactNode;
	classList?: string[];
}) => {
	return <div className={clsx("regex-box", classList)}>{children}</div>;
};
false;
false;
"
`;
